fastplotlib/widgets/image.py

```diff
@@ -192,6 +205,77 @@ def current_index(self) -> Dict[str, int]:
         """
         return self._current_index
 
+    @property
+    def num_img_dims(self) -> list[int]:
+        """
+
+        Returns:
+        -------
+        num_img_dims: list[int]
+            | ``list`` describing, for each array, how many dimensions are used to display a single image.
+        Can be either 2 or 3. Depending on whether the image is grayscale or RGB(A).
+        """
+        return self._num_img_dims
+
+    def _initialize_full_color_scheme(
+        self, color_scheme: Dict[int, str], data: list[np.ndarray]
+    ) -> Dict[int, str]:
+        """
+        Initializes the color scheme dictionary, populating it with a color scheme for each `array` which the
+        ImageWidget displays.
+
+        Returns:
+        -------
+        full_color_schemes: Dict[int, str]
+            | ``dict`` whose keys are indices (0, 1, 2, ...) for each array and values are "gray" or "rgb" depending on
+            how we want to display the images from this array
+        """
+        full_color_scheme = dict()
+        for k in range(len(data)):
+            if k not in color_scheme.keys():
+                full_color_scheme[k] = DEFAULT_IMAGE_FORMAT
+            elif k in color_scheme.keys():
+                curr_scheme = color_scheme[k].lower()
+                if curr_scheme not in RGB_DIM_MAP.keys():
+                    raise ValueError(
+                        f"Color Schemes can only be one of {list(RGB_DIM_MAP.keys())} "
+                        f"You passed in {color_scheme[k]} for one of the keys"
+                    )
+                else:
+                    full_color_scheme[k] = curr_scheme
+        return full_color_scheme
+
+    def _compute_and_validate_formats(self) -> list[int]:
```

Comment:
This is probably useful when `set_data()` is used.

This should take args, `data: List[<array_like>], rgb: List[bool]`

maybe rename this to `_get_n_scrollab_dims(data, rgb)`